src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -64,13 +64,13 @@
 
   (retrieve-ca-cert!
     [this localcacert]
-    (ca/retrieve-ca-cert! (get-in-config [:puppetserver :cacert])
-                          localcacert))
+    (let [settings (ca/config->ca-settings (get-config))]
```

Comment:
I think `ca/config->ca-settings` is rather heavyweight and none of its values should change during the service's lifetime. I think it would be good to compute it once during `init` and save it into the `context` then grab the service's context in these protocol implementations.

Reply:
See start and end lines [here](https://github.com/puppetlabs/puppetserver/blob/master/src/clj/puppetlabs/services/config/puppet_server_config_service.clj#L32-L37) for how we do that in the config service.